spec/end_to_end_spec.rb

```diff
@@ -0,0 +1,162 @@
+    end
+  end
+
+  context "meta suite run information" do
+    it "has version and legend information" do
+        expected_meta = <<-META
+Welcome to Gorgon #{Gorgon::VERSION}
+before job starts was called
+Loading environment and workers...\r                                  \rLegend:
+F - failure files count
+H - number of hosts that have run files
+W - number of workers running files
+
+
+Progress: ||
+
+        META
+        expect(@outputs[0]).to eq(expected_meta)
+    end
+  end
+
+  context "last hunk" do
+    it "has information for running after_job_finishes callback" do
+      expect(@outputs.last).to match(AFTER_RUNNING_REGEX)
+    end
+  end
+
+  context "exception test" do
+    it "has proper error output" do
+      actual_output = @outputs.grep /Stuff::Haha1/
+      expect(actual_output[0]).not_to be_nil
+      actual_output = actual_output[0].gsub(COLOR_REGEX, '')
+      actual_output = actual_output.gsub(AFTER_RUNNING_REGEX, '')
+      expected_output = <<-EXPECTED
+
+File 'test/unit/exception_test.rb' failed/crashed at '#{HOSTNAME}:1'
+Error:
+test_going_to_blow_up(Stuff::Haha1):
+RuntimeError: oh mah gawd
+    test/unit/exception_test.rb:10:in `test_going_to_blow_up'
+
+      EXPECTED
+
+      expect(actual_output).to eq(expected_output)
+    end
+  end
+
+  context "exception spec" do
+    it "has proper error output" do
+      actual_output = @outputs.grep /exception_spec/
+      expect(actual_output[0]).not_to be_nil
+      actual_output = actual_output[0].gsub(COLOR_REGEX, '')
+      actual_output = actual_output.gsub(AFTER_RUNNING_REGEX, '')
+      expected_output = <<-EXPECTED
+
+File 'spec/exception_spec.rb' failed/crashed at '#{HOSTNAME}:1'
+Test name: Exception spec raises: line 2
+RuntimeError
+Message: 
+	BOOOOM from spec!
+
+
+      EXPECTED
+
+      expect(actual_output).to eq(expected_output)
+    end
+  end
+
+  context "syntax error test" do
+    it "has proper error output" do
+      actual_output = @outputs.grep(/1_syntax_error_test/)
+      expect(actual_output[0]).not_to be_nil
+      actual_output = actual_output[0].gsub(COLOR_REGEX, '')
+      actual_output = actual_output.gsub(AFTER_RUNNING_REGEX, '')
+      actual_output = actual_output.gsub(PATH_REGEX, '')
+      expected_output = <<-EXPECTED
+
+File 'test/unit/1_syntax_error_test.rb' failed/crashed at '#{HOSTNAME}:1'
+Exception: test/unit/1_syntax_error_test.rb:9: syntax error, unexpected end-of-input, expecting keyword_end
+
+      EXPECTED
+      expect(actual_output).to eq(expected_output)
+    end
+  end
+
+  context "syntax error spec" do
+    it "has proper error output" do
+      actual_output = @outputs.grep(/1_syntax_error_spec/)
+      expect(actual_output[0]).not_to be_nil
+      actual_output = actual_output[0].gsub(COLOR_REGEX, '')
+      actual_output = actual_output.gsub(AFTER_RUNNING_REGEX, '')
+      actual_output = actual_output.gsub(PATH_REGEX, '')
```

Comment:
I'm not sure about this style of re-assigning the `actual_output` variable. We like to use the Arrange, Act, Assert pattern in our tests so I would be more comfortable if the steps were more distinct.